### PR TITLE
feat: introduce posibility to render series of bars for BarChart for buildChartOptions

### PR DESCRIPTION
**Pull request by ikondrat**
## What has changed and why?

This PR introduces options to render series of bars for function creation options for BarChart

It is organised as stacked diff commits, as solid requirement for BarChart 

It is better to review changes commit by commit
Starting from https://github.com/lightly-ai/lightly-studio/pull/1992/changes/b92ff99ca7f46b421a6df3c4a933d02d4d568761

## How has it been tested?

It should be enough to pass ci checks

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

---

<sub>Stack created with <a href="https://github.com/github/gh-stack">GitHub Stacks CLI</a> • <a href="https://gh.io/stacks-feedback">Give Feedback 💬</a></sub>

## New Features
- Added grouped bar charts with aligned categories, distinct series colors, and optional scrollable legends.
- Added number and percentage display modes, including filtered-versus-total counts.
- Improved tooltips with grouped labels, percentages, counts, and escaped content.
- Added stable category handling for duplicate or missing labels.
- Added deterministic colors that avoid collisions between series.

## Tests
- Expanded coverage for charts, tooltips, filtering, colors, and edge cases.

## Documentation
- Updated component stories to use current multi-select definitions.

### CODE REVIEW

File: lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    marker?: string;
+}
+
+const formatTooltipValue = (count: number, total: number): string => {
+    const percent = total > 0 ? ` (${formatPercent(count / total)})` : '';
+    return `<b>${count}</b>${percent}`;
+};
+
+/** Builds the ECharts tooltip formatter for grouped or standard bar charts. */
+export function buildChartTooltipFormatter(
```
Comment (LeonardoRosaa) [inline lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.ts]: We can use object destructuring to improve readability and maintainability. Suggestion:

```ts
export function buildChartTooltipFormatter({
  isGrouped,
  data,
  groupedSeries,
  totalCount,
  hasActiveFilter
}: {
  isGrouped: boolean;
  data: CategoryCount[];
  groupedSeries: CategoryCountSeries[];
  totalCount: number;
  hasActiveFilter: boolean;
}): (params: TooltipParam[]) => string
```

It suggested an inline type, but we can implement an interface.

File: lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    }
+];
+
+const param = (name: string, value: number, overrides: object = {}) => ({
+    name,
+    value,
+    ...overrides
+});
+
+describe('buildChartTooltipFormatter — standard (non-grouped)', () => {
```
Comment (LeonardoRosaa) [inline lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.test.ts:33]: Those tests are quite hard to understand. I'd suggest removing those `toContain` calls and writing the exact expected value. Example:

```ts
expect(result).toBe('<b>car</b>Count: 60')
```

File: lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.ts (hunk 1 above)
Comment (ikondrat) [reply lightly_studio_view/src/lib/components/BarChart/buildChartTooltipFormatter/buildChartTooltipFormatter.ts]: I would add interface, we can add jsdoc as well

File: lightly_studio_view/src/lib/components/BarChart/seriesColors/seriesColors.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// Same Tableau 10 palette used by the Histogram component.
+export const SERIES_COLORS = [
```
Comment (LeonardoRosaa) [inline lightly_studio_view/src/lib/components/BarChart/seriesColors/seriesColors.ts:2]: Could we reuse the logic that we have for annotations?

Comment (ikondrat) [reply lightly_studio_view/src/lib/components/BarChart/seriesColors/seriesColors.ts:2]: no it is different because here it is based by order and annotations colors based on label name

I would keep it separate for now, there is a separate task to make it customizable, we probably reuse logic we have for storing annotation colors

**Review by LeonardoRosaa — APPROVED** [review]
